packages/corvid-cli/src/apps/clone.js

```diff
@@ -38,50 +38,86 @@ function extractDataFromPublicUrl(parsedUrl) {
   return metasiteId;
 }
 
-async function extractMetasiteIdAndName(url, cookie) {
-  const publicSiteOrEditorUrl = normalize(url, { forceHttps: true });
-  const parsedUrl = new URL(publicSiteOrEditorUrl);
+function isInvalidUrl(url) {
+  try {
+    const normalizedUrl = normalize(url, { forceHttps: true });
+    const parsedUrl = new URL(normalizedUrl);
+    return !parsedUrl;
+  } catch (e) {
+    return true;
+  }
+}
+
+async function extractSiteData(url, cookie) {
+  const normalizedGivenUrl = normalize(url, { forceHttps: true });
+  const parsedUrl = new URL(normalizedGivenUrl);
   const siteList = await getUserSiteList(cookie);
   logger.addExtraData({ userSiteList: siteList });
 
   if (parsedUrl.hostname === editorDomain) {
     const metasiteId = extractDataFromEditorUrl(parsedUrl);
     const site = siteList.find(site => site.metasiteId === metasiteId);
 
-    return { metasiteId, siteName: site ? site.siteName : null };
+    return {
+      metasiteId,
+      siteName: site ? site.siteName : null,
+      siteBelongsToUser: !!site
+    };
   } else if (
     parsedUrl.hostname.endsWith(publicWixDomain) &&
     parsedUrl.pathname.startsWith(editorPath)
   ) {
     const metasiteId = extractDataFromPublicUrl(parsedUrl);
     const site = siteList.find(site => site.metasiteId === metasiteId);
 
-    return { metasiteId, siteName: site ? site.siteName : null };
+    return {
+      metasiteId,
+      siteName: site ? site.siteName : null,
+      siteBelongsToUser: !!site
+    };
   } else {
     const site = siteList.find(
       site =>
         site.publicUrl &&
-        normalize(site.publicUrl, { forceHttps: true }) ===
-          publicSiteOrEditorUrl
+        (normalize(site.publicUrl, { forceHttps: true }) ===
+          normalizedGivenUrl ||
+          normalizedGivenUrl.startsWith(
+            normalize(site.publicUrl, { forceHttps: true }) + "/"
+          ))
     );
 
     return {
       metasiteId: site ? site.metasiteId : null,
-      siteName: site ? site.siteName : null
+      siteName: site ? site.siteName : null,
+      siteBelongsToUser: !!site
     };
   }
 }
 
 async function clone(spinner, args, cookie) {
   spinner.start(chalk.grey(getMessage("Clone_Getting_Site_Data")));
   try {
-    const { metasiteId } = await extractMetasiteIdAndName(args.url, cookie);
+    if (isInvalidUrl(args.url)) {
+      throw new UserError(getMessage("Clone_Invalid_Url", { url: args.url }));
+    }
+
+    const { metasiteId, siteBelongsToUser } = await extractSiteData(
+      args.url,
+      cookie
+    );
 
     if (metasiteId == null) {
-      throw new Error(
+      throw new UserError(
```

Comment:
Clone_No_MetasiteId_Error should be renamed to something like Non_Wix_Url_Or_No_Permissions and the text should be changed accordingly

Reply:
done